### PR TITLE
Talent Captive Portal

### PR DESCRIPTION
**Pull request by dtgreiner**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch
- use a merge-commit or rebase strategy for PRs targeting the stable branch

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)

Updates for the Talent Captive Portal:
- Add Site Level config for number of courses the user is required to complete and expose this on the TalentLMS admin page
- Send user to captive portal if multiple LMS configs require training (if only one config requires training, send the user directly to that portal)
- Link to each course (if the user has a login for  the corresponding portal) on the captive portal
- Updates to phrasings on the captive portal

## Type of change
[//]: # (e.g., Bug fix, New feature, Code clean-up, Dependency update)
New feature

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [X] If adding a new endpoint / exposing data in a new way, I have:
  - [X] ensured the API can't leak data from other data sources
  - [X] ensured this does not introduce N+1s
  - [X] ensured permissions and visibility checks are performed in the right places
- [ ] I have updated the documentation (or not applicable)
- [ ] I have added spec tests (or not applicable)
- [x] I have provided testing instructions in this PR or the related issue (or not applicable)

### CODE REVIEW

File: app/controllers/admin/talentlms_controller.rb [hunk 1]
```diff
@@ -43,6 +48,20 @@ def destroy
       @config = config_scope.find(params[:id].to_i)
     end
 
+    private def set_site_config
+      @site_configs = site_config_scope.where(id: 1).first_or_create
```
Comment (eanders) [inline app/controllers/admin/talentlms_controller.rb:52]: Did you intend to leave `where(id: 1)` here?

File: app/views/admin/talentlms/index.haml [hunk 2]
```diff
@@ -6,63 +6,97 @@
...
-        %th Default
-        %th
-    %tbody
-      - @configs.each do |config|
-        - config.courses.each do |course|
+.h2 
+  Configuration
+.well
+  = simple_form_for @site_configs, url: update_site_config_admin_talentlms_path(@site_configs), method: :post, remote: true do |f|
+    = f.input :number_lms_courses_required, as: :select_two, collection: { "All": -1, "One of Any": 1 },  input_html: {class: :jRemoteSubmit}, label: 'TalentLMS Courses Completion Requirement'
```
Comment (eanders) [inline app/views/admin/talentlms/index.haml]: Can we move the value for `collection` into the config model?  We should call something like `@site_configs.class.available_completion_counts` (that's probably not quite right).  Having the options in the view makes them harder to enforce and to track down later.

**Review by eanders — COMMENTED** [review]
@dtgreiner this looks good.  Can you give me a tour (and maybe add some screenshots of the changes to the associated issue?)

File: app/controllers/admin/talentlms_controller.rb (hunk 1 above)
Comment (dtgreiner) [reply app/controllers/admin/talentlms_controller.rb:52]: This is how it's being referenced elsewhere in the app ([example](https://github.com/greenriver/hmis-warehouse/blob/dg/talent-captive-portal_6669/app/controllers/admin/configs_controller.rb#L30), [example](https://github.com/greenriver/hmis-warehouse/blob/dg/talent-captive-portal_6669/app/controllers/admin/de_duplication_controller.rb#L42), [example](https://github.com/greenriver/hmis-warehouse/blob/dg/talent-captive-portal_6669/spec/models/grda_warehouse/system_cohorts/system_cohort_spec.rb#L31)) so I matched to stay consistent. It does feel like removing the `.where(id: 1)` from each of these locations and going with `GrdaWarehouse::Config.first_or_create` would be better. Happy to adjust it in all of these spots if you feel it is worth doing.

File: app/views/admin/talentlms/index.haml (hunk 2 above)
Comment (dtgreiner) [reply app/views/admin/talentlms/index.haml]: Will add this in the next push to the PR

**Comment by dtgreiner** [comment]
### Updated TalentLMS Admin Page
Screenshot 2024-11-20 at 8 45 37 AM
Screenshot 2024-11-20 at 8 45 42 AM

### Updated Captive Portal
Screenshot 2024-11-20 at 8 46 15 AM

File: app/controllers/admin/talentlms_controller.rb (hunk 1 above)
Comment (eanders) [reply app/controllers/admin/talentlms_controller.rb:52]: Oh, I mis-read that, I was thinking `GrdaWarehouse::Config` was some sort of other Talent related config.  Huh, that's weird, but I think the issue is that you have to have something between the class and `.first_or_create`.  Let's leave this as is.  Thanks.

File: app/views/admin/talentlms/index.haml (hunk 2 above)
Comment (dtgreiner) [reply app/views/admin/talentlms/index.haml]: This has been included in [this commit](https://github.com/greenriver/hmis-warehouse/pull/4944/commits/8921e9abdb3166418ba55263e027c235c8a0cc52).

**Review by eanders — APPROVED** [review]
Thanks @dtgreiner !

**Comment by eanders** [comment]
@dtgreiner can you fix the conflicts and then squash merge?  Thanks!

**Pull request merged**